dns/nameserver.go

```diff
@@ -0,0 +1,85 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// Nameserver represents a DNS nameserver.
+type Nameserver struct {
+	// IPAddr is the IP address of the nameserver.
+	IP net.IP
+
+	// Port is the port of the nameserver.
+	Port uint16
+}
+
+// Addr returns the address of the nameserver as a string.
+func (n Nameserver) Addr() string {
+	return n.IP.String() + ":" + strconv.Itoa(int(n.Port))
+}
+
+// ParseNameserverAddr parses a nameserver address string into an IP and a port.
+//
+// It expects the `addr` to be in the format `ip` or `ip[:port]`. Where `ip` can be an IPv4 or an IPv6 address.
+func ParseNameserverAddr(addr string) (Nameserver, error) {
+	hostStr, portStr, err := parseHostAndPort(addr)
+	if err != nil {
+		return Nameserver{}, err
+	}
+
+	// Parse the host part into an IP
+	ip := net.ParseIP(hostStr)
+	if ip == nil {
+		return Nameserver{}, fmt.Errorf("invalid nameserver IP address: %s", hostStr)
+	}
+
+	if portStr == "" {
+		portStr = "53"
+	}
+
+	// Convert the port to an integer
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return Nameserver{}, fmt.Errorf("nameserver port is not a number: %w", err)
+	}
+
+	if port < 0 || port > 65535 {
+		return Nameserver{}, fmt.Errorf("nameserver port out of range: %d", port)
+	}
```

Comment:
Why not return it directly from `parseHostAndPort`?

Reply:
Done 👍🏻